Tests/ResponseTests.swift

```diff
@@ -323,6 +323,68 @@ class ResponseJSONTestCase: BaseTestCase {
     }
 }
 
+class ResponseJSONDecodableTestCase: BaseTestCase {
+    struct HTTPBinResponse: Decodable {
+        let headers: [String: String]
+        let origin: String
+        let url: String
+    }
+
+    func testThatResponseJSONReturnsSuccessResultWithValidJSON() {
+        // Given
+        let urlString = "https://httpbin.org/get"
+        let expectation = self.expectation(description: "request should succeed")
+
+        var response: DataResponse<HTTPBinResponse>?
+
+        // When
+        Alamofire.request(urlString, parameters: [:]).responseJSONDecodable { (resp: DataResponse<HTTPBinResponse>) in
+            response = resp
+            expectation.fulfill()
+        }
+
+        waitForExpectations(timeout: timeout, handler: nil)
+
+        // Then
+        XCTAssertNotNil(response?.request)
+        XCTAssertNotNil(response?.response)
+        XCTAssertNotNil(response?.data)
+        XCTAssertNotNil(response?.data)
```

Comment:
you are testing two times the data, is this intended or a typo?

Reply:
No, I must have copy pasted something.

Reply:
Removed in [9e374bf](https://github.com/Alamofire/Alamofire/pull/2265/commits/9e374bfdb2cd81577d89338025ced09efc4608c5).